arangod/Cache/PlainCache.cpp

```diff
@@ -247,6 +247,12 @@ bool PlainCache<Hasher>::freeMemoryWhile(
   for (std::size_t i = 0; i < n; ++i) {
     std::uint64_t index = (offset + i) % n;
 
+    // we can do a lot of iterations from here. don't check for
+    // shutdown in every iteration, but only in every 1000th.
+    if (index % 1024 == 0 && ADB_UNLIKELY(isShutdown())) {
+      break;
+    }
+
     auto [status, guard] = getBucket(Table::BucketId{index}, Cache::triesFast,
                                      /*singleOperation*/ false);
```

Comment:
Shouldn't we use the new `getBucket` function here? We already have the table ptr in hand...

Reply:
Yes, will adjust.